### PR TITLE
Support for performing key provisioning on any slot and backup key generation logic.

### PR DESCRIPTION
**Pull request by kashifpk**
Regarding https://github.com/NebraLtd/product-management/issues/180

**Checklist**

- [x] Tests added
- [x] Cleaned up commit history (rebase!)
- [x] Documentation added
- [x] Thought about variable and method names

### CODE REVIEW

**Review by ilyastrodubtsev — APPROVED** [review]

File: hm_pyhelper/miner_param.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def get_gateway_mfr_version() -> Version:
         raise GatewayMFRInvalidVersion(err_str).with_traceback(e.__traceback__)
 
 
-def get_gateway_mfr_command(sub_command: str) -> list:
+def get_gateway_mfr_command(sub_command: str, slot: int = 0) -> list:
```
Comment (pritamghanghas) [inline hm_pyhelper/miner_param.py:104]: it is alright for the time being, no need to change. But slot kind of goes against SWARM_KEY_URI as it always overrides the slot it provides.

File: hm_pyhelper/miner_param.py [hunk 2]
```diff
@@ -132,6 +132,9 @@ def get_gateway_mfr_command(sub_command: str) -> list:
         except (UnknownVariantException, UnknownVariantAttributeException) as e:
             LOGGER.warning(str(e) + ' Omitting --device arg.')
 
+        slot_str = f'slot={slot}'
```
Comment (pritamghanghas) [inline hm_pyhelper/miner_param.py:135]: slot is being ignore for backward compatible path on L123. I am in favor of totally removing backward compatible path.

File: setup.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 setup(
     name='hm_pyhelper',
-    version='0.13.36',
+    version='0.13.37',
```
Comment (pritamghanghas) [inline setup.py:6]: as none of the my comments will impact the current functionality. Given time critical nature of this change. we can take care of them in a separate ticket.

**Review by pritamghanghas — APPROVED** [review]

File: hm_pyhelper/miner_param.py (hunk 1 above)
Comment (kashifpk) [reply hm_pyhelper/miner_param.py:104]: Yes, I didn't want to change hardware definitions as they are used in other parts of code too. Ideally slot should not have been part of SWARM_KEY_URI.

File: hm_pyhelper/miner_param.py (hunk 2 above)
Comment (kashifpk) [reply hm_pyhelper/miner_param.py:135]: Yes, I'm not updating backward compatible code and also think it should be removed. Feel free to open up a ticket for it.

**Pull request merged**